packages/inertia/src/inertia.js

```diff
@@ -221,31 +278,33 @@ export default {
   },
 
   pushState(page) {
-    window.history.pushState({
-      cache: {},
-      ...page,
-    }, '', page.url)
+    this.page = page
+    window.history.pushState(page, '', page.url)
   },
 
-  replaceState(page, preserveState = false) {
-    window.history.replaceState({
-      ...{ cache: preserveState && window.history.state ? window.history.state.cache : {} },
-      ...page,
-    }, '', page.url)
+  replaceState(page) {
+    this.page = page
+    window.history.replaceState(page, '', page.url)
   },
 
-  restoreState(event) {
-    if (event.state) {
-      const page = this.transformProps(event.state)
+  handlePopstateEvent(event) {
+    if (event.state !== null) {
+      let page = event.state
       let visitId = this.createVisitId()
       return Promise.resolve(this.resolveComponent(page.component)).then(component => {
         if (visitId === this.visitId) {
+          this.page = page
           this.swapComponent({ component, page, preserveState: false }).then(() => {
```

Comment:
It could be relevant to use some async/await instead of nesting promises like that?

Reply:
If you wanted to submit a subsequent PR for this, I'd certainly consider it. I don't know what's all involved, so I'm going to keep it this way for this PR. 👍